### DISCUSSION TITLE
Rename module -> change `import a` to `import a as b`

### DISCUSSION THREAD

**Discussion by znd4** [microsoft/pylance-release#3488] [Enhancement]
**Is your feature request related to a problem? Please describe.**

I'd like to be able to easily autoimport a module (like `numpy`) and then rename it (e.g. to `np`) without having to move up to my import block

**Describe the solution you'd like**
(Sorry that this won't be precise / use correct terminology. I don't know the LSP super well)

Allow renaming a module. Assuming the new module name is valid / doesn't conflict with another module name (or any variable name?), change the import of the module to an `import x as y` (e.g. `import numpy as np`)

**Additional context**
I know the equivalent of this is implemented in `gopls`, but I'm not sure if the namespace collision problems are easier to solve for in go than in python?

**Comment by erictraut** [comment]
Pyright is focused on core type checking features. This is a language server enhancement request, so I'll transfer it to the pylance-release project so it gets the appropriate attention.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
